scenes/cslib_menu.txt

```diff
@@ -149,121 +167,202 @@
 *temp option_slot_id_9 0
 *temp option_slot_id_10 0
 
-*temp ignore 0
-*set ignore ((cur_page - 1) * items_per_page)
-*temp item_count 0
-*temp cur_option 1
-*temp cur_item "@{invert_order ${{prefix&\"_len\"}}|1}"
-*label _filter_loop
-*if (filter = "")
-	*goto _do_not_filter
-*if ({((prefix&"_")&(filter&"_"))&cur_item} = filter_value)
-	*if (filter_type = "NOT_MATCH")
-		*goto _acc
-*elseif (filter_type != "NOT_MATCH")
-	*goto _acc
-*label _do_not_filter
-*set item_count + 1
-*if (ignore < 1) and (cur_option <= items_per_page)
-	*set option_slot_id[cur_option] cur_item
-	*set cur_option + 1
-*if (ignore > 0)
-	*set ignore -1
-*label _acc
-*if (invert_order)
-	*if (cur_item > 1)
-		*set cur_item - 1
-		*goto _filter_loop
-*else
-	*if (cur_item < {prefix&"_count"})
-		*set cur_item + 1
-		*goto _filter_loop
-*if (item_count = 0) and not(allow_cancel)
-	*bug cslib_menu: menu construction failed as there were 0 matching items — consider allowing cancellation
+*temp total_item_count 0
+*temp cur_option_index 1
+*temp cur_item_index "@{const_invert_order ${{prefix&\"_count\"}}|1}"
```

Comment:
Is this a multi-replace in a string? Wow, we are pushing the envelope here. This is actually a very powerful operator.

Reply:
Yeah, basically a ternary (again).